### PR TITLE
update cocos engine v2.4.13 for openharmony 

### PR DESCRIPTION
**Pull request by zhaoyang57**
1. add frame lock implementation; 2. Upgrade to API10;  3.Add query network type interface implementation; 4. Change the supported device types to phones and tablets

### CODE REVIEW

**Comment by qiuguohua** [comment]
Please use English, including titles, submission logs, etc.

File: templates/js-template-link/frameworks/runtime-src/proj.openharmony/entry/build-profile.json5 [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "buildOption": {
     "externalNativeOptions": {
       "path": "../CMakeLists.txt",
-      "arguments": "-v arm-linux-ohos -DOHOS_STL=c++_shared -std=c++17 -DCOCOS_X_PATH=${COCOS_X_PATH} -DOPENHARMONY=1",
+      "arguments": "-v arm-linux-ohos -DOHOS_STL=c++_shared -std=c++17 -DCOCOS_X_PATH=../../cocos2d-x -DOPENHARMONY=1",
```
Comment (qiuguohua) [inline templates/js-template-link/frameworks/runtime-src/proj.openharmony/entry/build-profile.json5]: This template is used by the editor and does not need to be changed?

File: templates/js-template-link/frameworks/runtime-src/proj.openharmony/entry/src/main/ets/cocos/sys-ability-polyfill.js [hunk 2]
```diff
@@ -131,17 +131,14 @@ globalThis.getDeviceMotionValue = function () {
...
-        }).catch(err => { 
-            console.log('getDefaultNet error.' + JSON.stringify(err));
-            return -1; // 未获取到网络连接类型
-        })
-    ).catch(err => { 
-        console.log('getDefaultNet error.' + JSON.stringify(err));
-        return -1; // 网络未连接
-    });
+    let netHandle = connection.getDefaultNetSync();
+    if(netHandle && netHandle.netId != 0) {
```
Comment (qiuguohua) [inline templates/js-template-link/frameworks/runtime-src/proj.openharmony/entry/src/main/ets/cocos/sys-ability-polyfill.js]: One space is missing after ‘if’

File: cocos/platform/openharmony/OpenHarmonyPlatform.h [hunk 3]
```diff
@@ -78,6 +78,8 @@ class OpenHarmonyPlatform {
     static void onMessageCallback(const uv_async_t* req);
     static void timerCb(uv_timer_t* handle);
     void tick();
+    int64_t _prefererredNanosecondsPerFrame{NANOSECONDS_60FPS};
```
Comment (qiuguohua) [inline cocos/platform/openharmony/OpenHarmonyPlatform.h]: Can we put member variables together?

File: templates/js-template-default/frameworks/runtime-src/proj.openharmony/entry/src/main/module.json5 [hunk 4]
```diff
@@ -5,7 +5,8 @@
     "description": "$string:entry_desc",
     "mainElement": "EntryAbility",
     "deviceTypes": [
-      "default"
+      "phone",
```
Comment (qiuguohua) [inline templates/js-template-default/frameworks/runtime-src/proj.openharmony/entry/src/main/module.json5:8]: Has the default type been removed? If there are other types in the future that need to be modified by users?

Also, what type is rk3568?

File: templates/js-template-link/frameworks/runtime-src/proj.openharmony/entry/build-profile.json5 (hunk 1 above)
Comment (zhaoyang57) [reply templates/js-template-link/frameworks/runtime-src/proj.openharmony/entry/build-profile.json5]: reverted

File: cocos/platform/openharmony/OpenHarmonyPlatform.h (hunk 3 above)
Comment (zhaoyang57) [reply cocos/platform/openharmony/OpenHarmonyPlatform.h]: done

File: cocos/platform/openharmony/OpenHarmonyPlatform.cpp [hunk 5]
```diff
@@ -273,19 +273,33 @@ void OpenHarmonyPlatform::onSurfaceChanged(OH_NativeXComponent* component, void*
...
+}
+
 void OpenHarmonyPlatform::tick() {
         static std::chrono::steady_clock::time_point prevTime;
         static std::chrono::steady_clock::time_point now;
         static float dt = 0.f;
         static float dtSum = 0.f;
+        static double dtNS = NANOSECONDS_60FPS;
+        if (dtNS < static_cast<double>(_prefererredNanosecondsPerFrame)) {
+        std::this_thread::sleep_for(std::chrono::nanoseconds(
```
Comment (qiuguohua) [inline cocos/platform/openharmony/OpenHarmonyPlatform.cpp:290]: Need spaces?

**Pull request closed without merging**